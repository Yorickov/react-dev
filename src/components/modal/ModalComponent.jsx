import React, { Component } from 'react';
import Modal from './Modal.jsx';

class ModalComponent extends Component {
  state = { modal: false };

  toggle = () => {
    const { modal } = this.state;
    this.setState({ modal: !modal });
  };

  render() {
    const { modal } = this.state;

    return (
      <div>
        <button className="modal-open-button btn btn-danger" onClick={this.toggle}>Open</button>
        <Modal isOpen={modal}>
          <Modal.Header toggle={this.toggle}>Modal title</Modal.Header>
          <Modal.Body>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit
          </Modal.Body>
          <Modal.Footer>
            <button className="footer-close-button btn btn-primary" onClick={this.toggle}>Cancel</button>
          </Modal.Footer>
        </Modal>
      </div>
    );
  }
}

export default ModalComponent;
