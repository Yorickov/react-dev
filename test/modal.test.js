import React from 'react';
import ModalComponent from '../src/components/modal/ModalComponent.jsx';

it('ModalComponent', () => {
  const wrapper = mount(<ModalComponent />);
  expect(wrapper.find('.modal').hasClass('show')).toBeFalsy();
  expect(wrapper.render()).toMatchSnapshot();

  const button = wrapper.find('.modal-open-button');

  button.simulate('click');
  expect(wrapper.find('.modal').hasClass('show')).toBeTruthy();
  expect(wrapper.render()).toMatchSnapshot();

  wrapper.find('.header-close-icon').simulate('click');
  expect(wrapper.find('.modal').hasClass('show')).toBeFalsy();

  button.simulate('click');
  expect(wrapper.find('.modal').hasClass('show')).toBeTruthy();
  wrapper.find('.footer-close-button').simulate('click');
  expect(wrapper.find('.modal').hasClass('show')).toBeFalsy();
});
