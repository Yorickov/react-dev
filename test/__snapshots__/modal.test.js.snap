// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalComponent 1`] = `
<div>
  <button
    class="modal-open-button btn btn-danger"
  >
    Open
  </button>
  <div
    class="modal"
    style="display: none;"
  >
    <div
      class="modal-dialog"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title"
          >
            Modal title
          </div>
          <button
            aria-label="Close"
            class="header-close-icon"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <p
          class="modal-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit
        </p>
        <p
          class="modal-footer"
        >
          <button
            class="footer-close-button btn btn-primary"
          >
            Cancel
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModalComponent 2`] = `
<div>
  <button
    class="modal-open-button btn btn-danger"
  >
    Open
  </button>
  <div
    class="modal fade show"
    style="display: block;"
  >
    <div
      class="modal-dialog"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title"
          >
            Modal title
          </div>
          <button
            aria-label="Close"
            class="header-close-icon"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <p
          class="modal-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit
        </p>
        <p
          class="modal-footer"
        >
          <button
            class="footer-close-button btn btn-primary"
          >
            Cancel
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;
